### ISSUE TITLE
Node ids must be persistent accross database generation

### ISSUE THREAD

**Issue by paulgirard** [edgeryders/ryderex#9]
To allow building permalinks for user session we need permanent ids for nodes.
We suspect current node ids not to be persistent across database generation.

@aerugo can you tell us more about ids and is there a way to build a persistent key in your generation script ?

**Comment by aerugo** [comment]
Yes, you are right, the ID will not be persistent. I can come up with a way
to generate persistent IDs for all nodes by hashing some unique combination
of their persistent properties. I will try to get to that later this week.

**Comment by sim51** [comment]
Every node has a `discourse_id` and it's a primary key (if we prefix it by the platform name). 
So what I did : 
* The GraphQl API returns  Neo4j ids
* I added a functionin frontend to migrate the graph ids.

**Comment by aerugo** [comment]
You need to prefix it with the type of record too though I think if you
want it to be unique. There may be a user, topic and post with the same
discourse id. Also, the codes and annotations don’t have them, but not sure
if that matters

**Comment by sim51** [comment]
FYI, code & annotation have a `discourse_id`, I checked.
And you have right, I need to add as a prefix, the "table" name.

**Comment by aerugo** [comment]
Ah, I stand corrected!
